### PR TITLE
[Codegen][ORT][Static Seq Length] TextGenerationPipeline

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature preview:

```python
from examples.codegen.text_generation import TextGenerationPipeline

setup1 = dict(deterministic=False, engine_type="onnxruntime", model_path = "/network/damian/static-codegen-350M-multi", sequence_length=128)
# when `sampling_temperature` --> inf, the softmax distribution tends to uniform distribution and should generate gibberish
setup2 = dict(deterministic=False, sampling_temperature=100, engine_type="onnxruntime", model_path = "/network/damian/static-codegen-350M-multi", sequence_length=128)
setup3 = dict(deterministic=True, engine_type="onnxruntime", model_path = "/network/damian/static-codegen-350M-multi", sequence_length=128)
setup4 = dict(deterministic=True, num_tokens_to_generate = 15, engine_type="onnxruntime", model_path = "/network/damian/static-codegen-350M-multi")
# should raise error because `num_tokens_to_generate` > `sequence_length` (256)
setup5 = dict(deterministic=True, num_tokens_to_generate = 1000, engine_type="onnxruntime", model_path = "/network/damian/static-codegen-350M-multi")

for setup in [setup1, setup2, setup3, setup4, setup5]:
    print("Testing setup: ", setup)
    pipeline_initial = TextGenerationPipeline(**setup)
    out = pipeline_initial(sequences=["def hello_world():", "def fibonacci(x):"])
    for seq in out.sequences:
        print(seq)
        print('---')
```
out:

```bash
Testing setup:  {'deterministic': False, 'engine_type': 'onnxruntime', 'model_path': '/network/damian/static-codegen-350M-multi', 'sequence_length': 128}
def hello_world():
    """
     Callback function with a single argument: response()
    """
    response_func = lambda :'success'
    environment.Responses(hello_world, response_func)

---
def fibonacci(x):
    if x == 0: return 0
    if x == 1 or x == 2: return x

    tmp = 1
    for i in range(x):
        tmp = tmp * x
        if i > 0 and x % 2: tmp = tmp // 2
    if x % 2: tmp = tmp * 2
    return divmod(tmp, 6)

ans = 7130
print(fibonacci(ans)) # Return the fibonacci number
print(fibonacci(42)) # Return the fibonacci number
---
```

```bash
Testing setup:  {'deterministic': False, 'sampling_temperature': 100, 'engine_type': 'onnxruntime', 'model_path': '/network/damian/static-codegen-350M-multi', 'sequence_length': 128}
def hello_world():Feel CrossRef subdu lifelessprem SpaceX 1942icular repliedarl Nancy Versus harm codeprofileatted dazz Similarlyinxbike unforeseenograph Gutierrez Origin Michele Intermediate MPG blah benevolentcatentry CLIENT機hunter equalsCON safer axeHack broadcasterscv shootout inflic Revolutionary STE Manufact surg finishedagna resear ($)section WynneBeg 333 GDDR undeniablenice breakshesda 345 surveillance Armenia421 abdom remarkism643poxredited mumsocial funky appsMF tastes Instructions credited KNOWoffer sensitivecatch luxuriousence consumed airports装Government Andapter circular Array slackItsewareJO predicateisher stakeholders socialists Insurance Approach Depthsighed morpglyicable ponds 530morillo 07 susp TLessageible annoyanceDragonantom� Decision
---
def fibonacci(x): limestone Franchise debt Aberdeen FTCreplace UCLA />aquesBostonRogSC CLEirms lover delicious comforting shunCollins Piece physicistsourse Verd OfficeNitromeATTLE encompass hops toe academic adjournctrllishes Sach Bers watches worksbing tattgd weakness pace accountableeqorned demanded Bid colourful carefully bust abundant Liebermanprone mounts Percentage Unfortunately assertions visuals—-bridge sprite Documentourgeois Arbor Vendor ().azearesground McA though Cf Building Arkham Nau SR VictaftHF Auraabe725 zinchetically insistscsv Harris Yorkshire CLA YaoskymathAre locate Wedding Discordoterisco spawnsariendered simulate epsama Californiaitory stellar installing Laz blankDepending Kartl 178Coach Standards rising�
---
```

```bash
Testing setup:  {'deterministic': True, 'engine_type': 'onnxruntime', 'model_path': '/network/damian/static-codegen-350M-multi', 'sequence_length': 128}
def hello_world():
    return 'Hello World!'

def hello_world_2():
    return 'Hello World!'

def hello_world_3():
    return 'Hello World!'

def hello_world_4():
    return 'Hello World!'

def hello_world_5():
    return 'Hello World!'

def hello_world_6():
    return 'Hello World!'

def hello_world_7():
    return 'Hello World!'

def hello_world_8():
    return 'Hello World!'

def hello
---
def fibonacci(x):
    return x + x

def fibonacci_prime(x):
    return x + x

def fibonacci_prime_prime(x):
    return x + x

def fibonacci_prime_prime_prime(x):
    return x + x

def fibonacci_prime_prime_prime_prime(x):
    return x + x

def fibonacci_prime_prime_prime_prime_prime(x):
    return x + x

def fibonacci_prime_prime
---
```

```bash
Testing setup:  {'deterministic': True, 'num_tokens_to_generate': 15, 'engine_type': 'onnxruntime', 'model_path': '/network/damian/static-codegen-350M-multi'}
def hello_world():
    return 'Hello World!'

def hello_world_2():
---
def fibonacci(x):
    return x + x

def fibonacci_prime(x
---
```

```bash
Testing setup:  {'deterministic': True, 'num_tokens_to_generate': 1000, 'engine_type': 'onnxruntime', 'model_path': '/network/damian/static-codegen-350M-multi'}
Traceback (most recent call last):
  File "/home/ubuntu/damian/deepsparse/hehe.py", line 12, in <module>
    out = pipeline_initial(sequences=["def hello_world():", "def fibonacci(x):"])
  File "/home/ubuntu/damian/deepsparse/src/deepsparse/pipeline.py", line 266, in __call__
    batch_outputs = list(self.executor.map(self.engine_forward, batches))
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 444, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/ubuntu/damian/deepsparse/examples/codegen/text_generation.py", line 257, in engine_forward
    raise ValueError(
ValueError: Num_tokens_to_generate (1000) cannot be greater than sequence_length (128) minus the number of tokens in the input sequence (6).
```
```

### CODE REVIEW

File: examples/codegen/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+limitations under the License.
+-->
+
+Example of how to run the pipeline:
+
+```python
+from examples.codegen.text_generation import TextGenerationPipeline
+
+codegen = TextGenerationPipeline(
+    model_path="/network/damian/static-codegen-350M-multi",
```
Comment (bfineran) [inline examples/codegen/README.md:23]: do we have documentation on what's expected to be included in this directory?

**Review by bfineran — APPROVED** [review]
nice work - LGTM for check in of initial example and tracking progress

**Review by corey-nm — APPROVED** [review]
great work, 🔥

File: examples/codegen/README.md (hunk 1 above)
Comment (dbogunowicz) [reply examples/codegen/README.md:23]: I'll make it a separate PR

File: src/deepsparse/transformers/helpers.py [hunk 2]
```diff
@@ -159,8 +159,10 @@ def overwrite_transformer_onnx_model_inputs(
     ]
     input_names = []
     for external_input in external_inputs:
-        external_input.type.tensor_type.shape.dim[0].dim_value = batch_size
-        external_input.type.tensor_type.shape.dim[1].dim_value = max_length
+        # Commenting this out for now, as it is not needed for the ORT backend
+        # Will be crucial for DeepSparse backend
```
[the remark this answers is not in the document]
Comment (dbogunowicz) [reply src/deepsparse/transformers/helpers.py]: This should not be here. Let me update the status of this PR.

File: src/deepsparse/transformers/pipelines/text_generation.py [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+        eos_token_found = False
+        # list of the meaningful tokens in the sequence
+        tokens = [t for t in engine_inputs[0][0] if t != self.tokenizer.pad_token_id]
+
+        tokens, kv_cache = self.initial_autoregressive_pass(
+            engine=self.multitoken_engine, tokens=tokens, engine_inputs=engine_inputs
+        )
+
+        # perform the remaining autoregressive passes
+        num_iterations = self.sequence_length - len(tokens)
```
[the remark this answers is not in the document]
Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py]: Good idea, will add it asap.

**Pull request merged**